deltacat/storage/model/metafile.py

```
@@ -106,13 +109,19 @@ def list_revisions(
 
     @staticmethod
     def latest_revision(
-        revision_dir_path: str,
-        filesystem: pyarrow.fs.FileSystem,
-        success_txn_log_dir: str,
-        current_txn_start_time: Optional[int] = None,
-        current_txn_id: Optional[str] = None,
-        ignore_missing_revision: bool = False,
-    ) -> MetafileRevisionInfo:
+            revision_dir_path: str,
+            filesystem: pyarrow.fs.FileSystem,
+            success_txn_log_dir: str,
+            current_txn_start_time: Optional[int] = None,
+            current_txn_id: Optional[str] = None,
+            ignore_missing_revision: bool = False,
+    ) -> Optional[MetafileRevisionInfo]:
+        """
+        Fetch latest revision of a metafile, or return None if no revisions exist
+        :param revision_dir_path: root path of directory for metafile
+        :param ignore_missing_revision: if True, will return MetafileRevisionInfo.undefined() on no revisions
+        :raises ValueError if no revisions are found AND ignore_missing_revision=False
+        """
```

having it return optional while also still having an 'ignore_missing_revision' feels like we're giving too many ways to do the same thing. I'd get rid of the flag and just return the optional.

---

In other words, leave it to the caller to decide whether or not they need to raise an exception? That may create a bit more code duplication overall, but is probably fine (this is just following the convention carried over from Ray Data right now).